chapter_philosophy_and_refactoring.asciidoc

```diff
@@ -593,6 +594,7 @@ class HomePageTest(TestCase):
 ----
 ====
 
+// JAN: I must say I have very mixed feelings about this way of testing. I get your point about testing plain HTML strings - makes complete sense. But I don't think using self.assertTemplateUsed(response, "home.html") makes it much better. We go from HTML implementation details (self.assertContains(response, "<title>To-Do lists</title>")) to Django implementation details (self.assertTemplateUsed(response, "home.html")). This test will pass whenever I'm using "home.html" template. It may be empty or complete garbage - this test will pass as long as the naming is correct. On the other hand, if I change the name of template, I'll have to update the test. Logic and ouput might stay completely the same. (I wouldn't be surprised if something like this would happen also when changing template engine). In other words, this test is not resistant to refactoring while also not protecting much against regression. Therefore, I'd consider it as low quality test - it won't protect me much while posibly failing when things still work exactly as expected. I know that there's not much breathing room there to actually change much. Nevertheless, I think it's worth bringing this up. (I derived this approach for test quality evaluation from this book: https://www.manning.com/books/unit-testing?query=%20Vladimir%20Khorikov)
```

Comment:
hmmm good point.

what do you tend to do for unit tests of GET endpoints, that just render templates?

Reply:
OK fair enough.  Being tightly coupled to Django is one of the main drawbacks of the book really, and it's hard to avoid.  The second book is the antidote!